src/D2L.CodeStyle.Analyzers/Common/MutabilityInspector.cs

```diff
@@ -75,84 +75,94 @@ public sealed class MutabilityInspector {
 		/// </summary>
 		/// <param name="type">The type to determine mutability for.</param>
 		/// <returns>Whether the type is mutable.</returns>
-		public bool IsTypeMutable(
+		public MutabilityInspectionResult InspectType(
 			ITypeSymbol type,
 			MutabilityInspectionFlags flags = MutabilityInspectionFlags.Default
 		) {
 			var typesInCurrentCycle = new HashSet<ITypeSymbol>();
-			var result = IsTypeMutableRecursive( type, flags, typesInCurrentCycle );
+			var result = InspectTypeRecursive( type, flags, typesInCurrentCycle );
 			return result;
 		}
 
-		private bool IsTypeMutableRecursive(
+		private MutabilityInspectionResult InspectTypeRecursive(
 			ITypeSymbol type,
 			MutabilityInspectionFlags flags,
 			HashSet<ITypeSymbol> typeStack
 		) {
-			if( type is IErrorTypeSymbol  || type == null ) {
-				throw new Exception( $"Type '{type}' cannot be resolved. Please ensure all dependencies " 
+			if( type is IErrorTypeSymbol || type == null ) {
+				throw new Exception( $"Type '{type}' cannot be resolved. Please ensure all dependencies "
 					+ "are referenced, including transitive dependencies." );
 			}
 
 			if( IsTypeKnownImmutable( type ) ) {
-				return false;
+				return MutabilityInspectionResult.NotMutable();
 			}
 
 			if( type.TypeKind == TypeKind.Array ) {
-				return true;
+				return MutabilityInspectionResult.Mutable( null, type.GetFullTypeName(), MutabilityTarget.Type, MutabilityCause.IsAnArray );

```

Comment:
The `null` names here are properly set by their the analyzer or by `InspectMemberRecursive`. We don't have the information here, but one level up the stack has it.